Migrate Studio Self hosted docs to dvc.org (#4344)
iterative/dvc.org: content/docs/studio/selfhosted/configuration/index.md

# Configuration

> Studio has a unified YAML configuration file, often referred to as `values.yaml`

Reviewer: ```suggestion
Like most Helm charts, Studio uses a standardized unified YAML configuration file, often referred to as `values.yaml`
```

Author: I think it's better not to mention Helm, so that AMI users don't get confused and/or worry about the underlying stack.

Reviewer: But it's impossible to avoid - no matter if we use AMI or BYOC k8s, it's a shared step - users will have to be aware of the helm and `values.yaml` file to configure studio. It's impossible to skip configuration when installing studio, no matter the infra - because of networking and SCM provider configuration at the very least, right?

in the documentation.

## Updating the configuration

> To update the configuration and reload Studio, follow the instructions according

Reviewer: ```suggestion
To update the configuration and reload Studio (to apply changes), follow the instructions according
```

to your installation method:

> <toggle>

Reviewer: I wouldn't do. a toggle here. We're using helm commands so configuration flow is unified, I would only add an admon/note that you have to run the command from a machine with kubectl access to the cluster, so if you're using an AMI (specifically), you may have to SSH into the instance.

no need tor tabs then

<tab title="AMI">

1. Connect with SSH to your EC2 instance

> ```shell

Reviewer 2: here and everywhere else - please use `cli` instead of `shell` - it will highlight better, and has other nice features (e.g. not including the $ when you are copy pasting ,etc)

Reviewer: @shcheklein do we have a nice way to generically add a button to copy contents to clipboard. it is almost necessary for such installation guides

Reviewer 3: We don't and we had a discussion on this here: https://github.com/iterative/gatsby-theme-iterative/issues/181.

$ ssh ubuntu@my-ec2-instance
```

2. Update Studio's `values.yaml` config file

**Example: Updating the hostname**

Merge the existing `values.yaml` file with the following contents:

```yaml
global:
  host: studio.company2.com
```

3. Reload Studio

```shell
$ helm upgrade --wait studio iterative/studio --namespace studio -f values.yaml
```

</tab>

<tab title="Helm">

1. Update Studio's `values.yaml` config file

**Example: Updating the hostname**

Merge the existing `values.yaml` file with the following contents:

```yaml
global:
  host: studio.company2.com
```

2. Reload Studio

```shell
$ helm upgrade --wait studio iterative/studio --namespace studio -f values.yaml
```

</tab>

</toggle>

## More configuration options

The previous chapter provided a basic example of updating the Studio
configuration. In most cases, you'll likely need to do more configuration to set
up Studio to your needs.

> Studio integrates with several Git forges, each of which are configured

Reviewer: ```suggestion
Studio integrates with several Git forges (also referred to as SCM providers), each of which are configured
```

> differently:

Reviewer: ```suggestion
differently. For any Git forge type you'll need to set up a Studio application prior to installation, and also configure a dedicated block under  `scmProviders` in the `studio-values.yaml` file:
```

- [GitHub](/doc/studio/selfhosted/configuration/github)
- [GitLab](/doc/studio/selfhosted/configuration/gitlab)

> 

Reviewer: bitbucket missing? followup?

> Additionally, you may want be interested in the following:

Reviewer: ```suggestion
To allow secure access to Studio, we highly recommend setting up a valid TLS certificate. To set this up, check out the [TLS certificates section](/doc/studio/selfhosted/configuration/tls)
```
